CONTRIBUTING.md

```diff
@@ -214,6 +214,52 @@ section refers to settings from that plugin.
    "shellformat.useEditorConfig": true,
   ```
 
+### Activation scripts
+
+Flox activations invoke a series of scripts
+which begins with the `activate` script
+as maintained in the `assets/activation-scripts` subdirectory.
+The process of developing these scripts is highly iterative,
+and it can be challenging to follow the sequence of scripts
+as invoked in different contexts.
+
+To make debugging easier,
+we have added statements to the top and bottom of each script
+like the following:
+
+```bash
+@coreutils@/bin/test -z "$FLOX_TRACE" || "$FLOX_TRACE" "$_activate_d/attach-inplace.bash" START
+...
+@coreutils@/bin/test -z "$FLOX_TRACE" || "$FLOX_TRACE" "$_activate_d/attach-inplace.bash" END
+```
+
+These statements are written this way
+expressly so as to remain compatible with all shells,
+and to run as efficiently as possible when not performing a trace.
+
+Enable tracing with one of the following:
+
+1. set `FLOX_TRACE` to a non-empty value
```

Comment:
suggestion: I might only be crown prince of bieke shedding, but a more specific name for this (`FLOX_ACTIVATION_TRACE`) might be more appropriate, given that it is independent of flox' tracing and specific to activations.

Reply:
Agreed, was expecting this.  😄  Wanted to have that conversation.

Reply:
Renamed the variable and logged #2554 to get the trace feature merged separately.